rust/src/lib/nispor/route_rule.rs

```diff
@@ -1,30 +1,66 @@
-use crate::{RouteRuleEntry, RouteRules};
+use log::warn;
 
-pub(crate) fn get_route_rules(np_rules: &[nispor::RouteRule]) -> RouteRules {
+use crate::{AddressFamily, RouteRuleEntry, RouteRules};
+
+const SUPPORTED_STATIC_ROUTE_PROTOCOL: [nispor::RouteProtocol; 3] = [
+    nispor::RouteProtocol::Boot,
+    nispor::RouteProtocol::Static,
+    nispor::RouteProtocol::UnSpec,
```

Comment:
Any reason why we have additional `UnSpec` route? 
And please  remove duplicate `SUPPORTED_ROUTE_PROTOCOL` and `SUPPORTED_STATIC_ROUTE_PROTOCOL` against `nispor::route`.

Reply:
> * At the top `RouteRules::pre_edit_clean()`, we always set `RouteRuleEntry.family` to correct value and raise error if family is incorrect according to `ip_from` or `ip_to`. And also convert `any` to `0.0.0.0/0` or `::/0`.

I do not agree with this. Nmstate should not convert `any` to `0.0.0.0/0` or `::/0`. It is redundant and could confuse users as the standard way to set routing policy `from all to all` is not setting the ip-from/ip-to fields. If we auto-complete the missing fields with `0.0.0.0/0` or `::/0` it will also be shown in the current state. IMO, this shouldn't be like that. Why do you think Nmstate should add `0.0.0.0/0`?

> 
>     * In the `gen_nm_ip_setting()`, we can only pass the rules for certain family. like:
> 
> 
> ```rust
>     gen_nm_ipv6_setting(
>         base_iface.ipv6.as_ref(),
>         routes,
>         rules.filter(|r| r.family == Some(AddressFamily::IPv6)),
>         nm_conn,
>     )?;
> ```

I agree here, let me work on this. Thanks!

> 
>     * Instruct nispor plugin to always include address family.

This is already done, please check `rust/src/lib/nispor/route_rule.rs:53`. Thanks!

Reply:
> Any reason why we have additional `UnSpec` route? And please remove duplicate `SUPPORTED_ROUTE_PROTOCOL` and `SUPPORTED_STATIC_ROUTE_PROTOCOL` against `nispor::route`.

This due to a bug in NetworkManager that I plan to fix. All route rules added using NetworkManager they have UnSpec protocol set. It shouldn't be like that. I already discussed this with Thomas and will send a patch but meanwhile we should allow UnSpec protocol rules.

Therefore, `SUPPORTED_ROUTE_PROTOCOL` and `SUPPORTED_STATIC_ROUTE_PROTOCOL` are not duplicated from the ones at `route.rs`